File: baekyeonsu/PGS/Lv. 1/문자열 나누기.js
function solution(s) {
  let prev = '';
  let lastIndex = 0;
  let numbers = {
    same: 0,
    different: 0,
  };
  let result = 0;

  s.split('').forEach((word, index) => {
    if (prev === word) {
      numbers.same += 1;
    } else if (prev === '') {
      numbers.same += 1;
      prev = word;
    } else {
      numbers.different += 1;
    }

    if (numbers.same === numbers.different) {
      prev = '';
      numbers.same = 0;
      numbers.different = 0;
      result += 1;
      lastIndex = index;
    }
  });

  if (lastIndex !== s.length - 1 || s.length === 1) {
    result += 1;
  }

  return result;
}

File: kimsei/PGS/LEVEL1/문자열_나누기.java
/**
 * 풀이 시간 : 7분
 */
class 문자열_나누기 {
    public int solution(String s) {
        int answer = 0;
        char temp = ' ';
        int[] x = new int[2];   // [0] : same, [1] : different
        char[] array = s.toCharArray();
        for (char c : array) {
            if (x[0] == 0) {
                x[0]++;
                answer++;
                temp = c;
                continue;
            }
            if (temp == c) {
                x[0]++;
            } else {
                x[1]++;
            }
            if (x[0] == x[1]) {
                x = new int[2];
            }
        }
        return answer;
    }
}